src/server/Model/Anonymization/Anonymizer.cs

```diff
@@ -141,24 +141,54 @@ static class Fuzzer
             prop.SetValue(record, composite.GetConsistentHashCode().ToString());
         };
 
-        public static readonly Actor DateTime = (record, prop, salt, pepper, parameters) =>
+        public static readonly Actor DateTimeActor = (record, prop, salt, pepper, parameters) =>
         {
             var rand = new Random(salt.GetHashCode());
-            var value = (DateTime)prop.GetValue(record);
+            var value = (System.DateTime)prop.GetValue(record);
             var shift = rand.Next(parameters.LowerBound, parameters.UpperBound);
 
-            prop.SetValue(record, parameters.DateShifter(value, shift));
+            // Clamp shift if needed
+            if (shift > 0 && value > System.DateTime.MaxValue.AddDays(-shift))
```

Comment:
please write more comments explaning this

Reply:
done